src/routes/categories/containers/CategoriesPage.js

```diff
@@ -1,66 +1,95 @@
+import CATEGORIES_ENDPOINT from 'modules/endpoint/endpoints/categories'
 import LANGUAGES_ENDPOINT from 'modules/endpoint/endpoints/languages'
-import PAGE_ENDPOINT from 'modules/endpoint/endpoints/pages'
-
-import Hierarchy from '../Hierarchy'
-import PageModel from 'modules/endpoint/models/PageModel'
-import { setLanguageChangeUrls } from 'modules/language/actions/setLanguageChangeUrls'
+import LOCATION_ENDPOINT from 'modules/endpoint/endpoints/location'
+import CategoryModel from 'modules/endpoint/models/CategoryModel'
 import LanguageModel from 'modules/endpoint/models/LanguageModel'
+import LocationModel from 'modules/endpoint/models/LocationModel'
+import { setLanguageChangeUrls } from 'modules/language/actions/setLanguageChangeUrls'
+
+import Breadcrumb from 'routes/categories/components/Breadcrumbs'
+import PdfButton from 'routes/categories/components/PdfButton'
+import Page from '../components/Page'
+import CategoryTiles from '../components/CategoryTiles'
+import TitledContentList from '../components/CategoryList'
 
 /**
- * Matching the route /<location>/<language>*
+ * Displays a CategoryTable, CategoryList or a single category matching the route /<location>/<language>*
  */
 class CategoriesPage extends React.Component {
   static propTypes = {
-    location: PropTypes.string.isRequired,
+    categories: PropTypes.arrayOf(PropTypes.instanceOf(CategoryModel)).isRequired,
+    locations: PropTypes.arrayOf(PropTypes.instanceOf(LocationModel)).isRequired,
     languages: PropTypes.arrayOf(PropTypes.instanceOf(LanguageModel)).isRequired,
+    location: PropTypes.string.isRequired,
     language: PropTypes.string.isRequired,
     path: PropTypes.string,
-    pages: PropTypes.instanceOf(PageModel).isRequired
-  }
-
-  getParentPath () {
-    return `/${this.props.location}/${this.props.language}`
+    categoryId: PropTypes.string,
+    dispatchLanguageChangeUrls: PropTypes.func.isRequired,
+    dispatchRedirectToUrl: PropTypes.func.isRequired
   }
 
+  /**
+   * If category is defined, we want to redirect to the page with the given id,
+   * else we just dispatch the language change urls here
+   */
   componentDidMount () {
+    if (this.props.categoryId) {
+      this.props.dispatchRedirectToUrl(this.getRedirectUrl(this.props.categories, this.props.categoryId))
+      return
+    }
     this.setLanguageChangeUrls(this.props.path)
   }
 
-  // we must not call dispatch in componentWillUpdate or componentDidUpdate
+  /**
+   * Generates the url of the category with the given id
+   * @param categories The categories to search in
+   * @param id The id of the category to search for
+   * @returns {string} The url of the category
+   */
+  getRedirectUrl = (categories, id) => {
+    const category = CategoryModel.getCategoryById(categories, id)
+    if (!category) {
+      // todo show an error that the page has not been found, redirect to the notFound route
+    }
+    return `/${this.props.location}/${this.props.language}/${category.url}`
```

Comment:
When calculating this redirect url do we really have the correct language? Did you test the language change?

Reply:
yeah language change works fine